macOS/DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/VPNAppEventsHandler.swift

```diff
@@ -32,69 +32,79 @@ import Subscription
-#else
-    private let legacyAgentBundleID = "HKE973VLUW.com.duckduckgo.macos.browser.network-protection.system-extension.agent"
-    private let legacySystemExtensionBundleID = "com.duckduckgo.macos.browser.network-protection-extension"
-#endif // DEBUG || REVIEW || RELEASE
-#endif // NETP_SYSTEM_EXTENSION
+    typealias FeatureFlagOverridesPublisher = AnyPublisher<(FeatureFlag, Bool), Never>
 
     // MARK: - Feature Gatekeeping
 
-    private let featureGatekeeper: VPNFeatureGatekeeper
-    private let uninstaller: VPNUninstalling
-    private let defaults: UserDefaults
     private var cancellables = Set<AnyCancellable>()
+    private let defaults: UserDefaults
+    private let featureGatekeeper: VPNFeatureGatekeeper
+    private let loginItemsManager: LoginItemsManaging
+
+    // MARK: - Initializers
 
     init(featureGatekeeper: VPNFeatureGatekeeper,
-         featureFlagOverridesPublishingHandler: FeatureFlagOverridesPublishingHandler<FeatureFlag>,
-         uninstaller: VPNUninstalling = VPNUninstaller(),
+         featureFlagOverridesPublisher: FeatureFlagOverridesPublisher,
+         loginItemsManager: LoginItemsManaging,
          defaults: UserDefaults = .netP) {
 
         self.defaults = defaults
-        self.uninstaller = uninstaller
         self.featureGatekeeper = featureGatekeeper
+        self.loginItemsManager = loginItemsManager
 
-        subscribeToFeatureFlagOverrideChanges(featureFlagOverridesPublishingHandler: featureFlagOverridesPublishingHandler)
+        subscribeToFeatureFlagOverrideChanges(featureFlagOverridesPublisher)
     }
 
     /// Call this method when the app finishes launching, to run the startup logic for NetP.
     ///
     func applicationDidFinishLaunching() {
-        let loginItemsManager = LoginItemsManager()
-
-        Task { @MainActor in
-            restartNetworkProtectionIfVersionChanged(using: loginItemsManager)
-        }
+        loginItemsControlCheckpoint(canRestart: true)
     }
 
-    private func restartNetworkProtectionIfVersionChanged(using loginItemsManager: LoginItemsManager) {
-        // We want to restart the VPN menu app to make sure it's always on the latest.
-        restartNetworkProtectionMenu(using: loginItemsManager)
+    func applicationDidBecomeActive() {
+        loginItemsControlCheckpoint(canRestart: false)
     }
 
-    private func restartNetworkProtectionMenu(using loginItemsManager: LoginItemsManager) {
-        guard loginItemsManager.isAnyEnabled(LoginItemsManager.networkProtectionLoginItems) else {
-            return
+    // MARK: - Login Item Control Checkpoints
+
+    /// Checks whether the VNP login items need to be disabled
+    ///
+    private func loginItemsControlCheckpoint(canRestart: Bool) {
+        Task { @MainActor [loginItemsManager] in
+            guard loginItemsManager.isAnyEnabled(LoginItemsManager.vpnLoginItems) else {
+
+                return
+            }
+
+            switch try? await featureGatekeeper.canStartVPN() {
```

Comment:
I set up a scenario where the login items are enabled, but I have no Privacy Pro account (which is plausible if something happens to the user's keychain, etc). When this happens, the `canStartVPN()` call throws an error instead of returning false, but because of the `try?` here it's actually the `.none` case that gets executed.

What do you think about running `disableLoginItem` even in the `none` case? What would happen is that the OS would throw `kSMErrorJobNotFound` for anyone who doesn't already have them running, but we aren't pixeling this logic so it wouldn't harm any metrics, and it would guarantee that the login item gets cleaned up in any case where there's no valid entitlement.